### PR TITLE
Fixed bug with updating user settings

### PR DESCRIPTION
**Pull request by milosevicj**
recreated in DB after every change instead of just updating them in DB)

### CODE REVIEW

File: src/main/java/eu/execom/labs/teamodoro/service/UserSettingsService.java [hunk 1]
```diff
@@ -23,6 +27,9 @@ public UserSettings getUserSettingsByUserId(Long userId) {
     @Transactional
     public void updateUserSettings(Long userId, UserSettings userSetings) {
         User user = userRepository.findOne(userId);
-        user.setSettings(userSetings);
+        long settingId = user.getSettings()
+                             .getId();
+        userSetings.setId(settingId);
+        userSettingsRepository.save(userSetings);
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/labs/teamodoro/service/UserSettingsService.java]: Does anybody know if there is a better way to do this?

Comment (dufajlusi) [inline src/main/java/eu/execom/labs/teamodoro/service/UserSettingsService.java]: userSettings.setId(userRepository.findOne(userId).getSettings().getId());
userSettingsRepository.save(userSettings);

Comment (stanislavgrujic) [inline src/main/java/eu/execom/labs/teamodoro/service/UserSettingsService.java]: What you propose is just a code realigning. 
I meant if anybody has an idea if there is a better way to update a database entity which is detached from the (hibernate) session than loading that entity just to pick up it's id (and as such load it to hibernate session) and then save it. Anyone?
e.g., Would it work this way (probably not, but you'll get an idea):
```
User user = userRepository.findOne(userId);
user.setUserSettings(userSettings);
userRepository.save(user);
```

Comment (PavleP) [inline src/main/java/eu/execom/labs/teamodoro/service/UserSettingsService.java]: I would like also to explore...

Comment (dufajlusi) [inline src/main/java/eu/execom/labs/teamodoro/service/UserSettingsService.java]: maybe cascade type should be changed from persist to merge?

Comment (PavleP) [inline src/main/java/eu/execom/labs/teamodoro/service/UserSettingsService.java]: With cascade type `merge` proposed by Dragan we will not have to load userSetting to bring it in persistent state. Method body should be:
        User user = userRepository.findOne(userId);
        userSettings.setId(user.getSettings().getId());
        user.setSettings(userSettings);
        userRepository.save(user);

Comment (PavleP) [inline src/main/java/eu/execom/labs/teamodoro/service/UserSettingsService.java]: Another possibility (also one reading from database) would be to pass in the path `userSettingsId`. Then we can keep `persist` cascade type and have a method:
```
@Transactional
public void updateUserSettings(Long userSettingsId, UserSettings userSettings) {
    UserSettings us = userSettingsRepository.findOne(userSettingsId);
    us.setLongBrakeDuration(userSettings.getLongBrakeDuration());
    us.setShortBrakeDuration(userSettings.getShortBrakeDuration());
    us.setPomodoroDuration(userSettings.getPomodoroDuration());
}
```

**Pull request merged**